File: algo/state.py
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()
state = {"message": "Hello, world!"}

class Message(BaseModel):
    new_message: str

@app.get("/message")
def get_message():
    return state

@app.post("/message")
def set_message(msg: Message):
    state["message"] = msg.new_message
    return state

# 	•	状態（state） = グローバルな message
# 	•	命令（command） = 関数呼び出し（GET/POST風）
# 	•	WebフレームワークなしでもRESTの設計思想を再現可能

# ⸻

File: algo/state.c
#include <stdio.h>
#include <string.h>

char message[100] = "Hello, world!";

void get_message() {
    printf("GET /message: %s\n", message);
}

void post_message(const char *new_msg) {
    strncpy(message, new_msg, sizeof(message) - 1);
    message[sizeof(message) - 1] = '\0';
    printf("POST /message: %s\n", message);
}

int main() {
    get_message();
    post_message("ぼっち・ざ・ホロリス！");
    get_message();
    return 0;
}